hack/check-copyright-year.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Copyright 2021 the Pinniped contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Check if copyright statements include the current year
+#
+files=$(git diff --cached --name-only)
+year=$(date +"%Y")
+
+for f in $files; do
+    head -10 $f | grep -i copyright 2>&1 1>/dev/null || continue
+
+    if ! grep -i -e "copyright.*$year" $f 2>&1 1>/dev/null; then
+        missing_copyright_files="$missing_copyright_files $f"
+    fi
+done
+
+if [ -n "$missing_copyright_files" ]; then
```

Comment:
I think `[[` is preferred over `[` since the former's behavior is more controlled. I dunno if you need to go back and make a special commit to fix that, but just wanted to pass that info off: https://google.github.io/styleguide/shellguide.html#test----and---.